app/javascript/crayons/MultiInput/MultiInput.jsx

```diff
@@ -0,0 +1,117 @@
+import { h, Fragment } from 'preact';
+import { useRef, useState } from 'preact/hooks';
+
+const KEYS = {
+  ENTER: 'Enter',
+  COMMA: ',',
+  SPACE: ' ',
+};
+// TODO: think about how this may change based on
+// a different usage. We may want this to be passed in as a prop.
+const ALLOWED_CHARS_REGEX = /([a-zA-Z0-9@.])/;
+
+/**
+ * Component allowing users to add multiple inputs that gets dislpayed as pills
+ *
+ * @param {Object} props
+ * @param {string} props.placeholder Input placeholder text
+ */
+
+export const MultiInput = ({ placeholder }) => {
+  const inputRef = useRef(null);
+
+  const [items, setItems] = useState([]);
+
+  const handleBlur = ({ target: { value } }) => {
+    addItemToList(value);
+    clearSelection();
+  };
+
+  const handleKeyDown = (e) => {
+    switch (e.key) {
+      case KEYS.SPACE:
+      case KEYS.ENTER:
+      case KEYS.COMMA:
+        e.preventDefault();
+        addItemToList(e.target.value);
+        clearSelection();
+        break;
+      default:
+        if (!ALLOWED_CHARS_REGEX.test(e.key)) {
+          e.preventDefault();
+        }
+    }
+  };
+
+  const handleDestructiveClick = (clickedItem) => {
+    const newArr = items.filter((item) => item !== clickedItem);
+    setItems(newArr);
+  };
+
+  const addItemToList = (value) => {
+    // TODO: we will want to do some validation here based on a prop
+    if (value.trim().length > 0) {
+      setItems([...items, value]);
+    }
+  };
+
+  const clearSelection = () => {
+    inputRef.current.value = '';
+  };
+
+  const listItems = items.map((item, index) => (
+    <li
+      key={index}
+      class="c-input--multi__selection-list-item w-max"
+      style="order: 1;"
+    >
+      <div role="group" aria-label="two" class="flex mr-1 mb-1 w-max">
+        <button
+          class="c-pill c-pill--action-icon c-pill--action-icon--destructive"
+          type="button"
+          aria-disabled="false"
+        >
+          {item}
+          <svg
```

Comment:
When testing voice over on this component, I realized that I did not add an event handle for KEYS.DELETE as it seemed a little more complicated so I'll tackle it on the next PR. I _think_ everything else should be accessible based on my testing but would appreciate another set of eyes please.